spyder/plugins/run/container.py

```diff
@@ -0,0 +1,858 @@
+            dirname = dirname.replace("'", r"\'").replace('"', r'\"')
+
+            last_executor = last_executor_name
+            if last_executor is None:
+                last_executor = self.get_last_used_executor_parameters(uuid)
+                last_executor = last_executor['executor']
+            run_comb = (extension, context)
+            if (last_executor is None or
+                    not self.executor_model.executor_supports_configuration(
+                        last_executor, run_comb)):
+                last_executor = self.executor_model.get_default_executor(
+                    run_comb)
+            executor_metadata = self.executor_model[
+                ((extension, context), last_executor)]
+            ConfWidget = executor_metadata['configuration_widget']
+
+            conf = {}
+            if ConfWidget is not None:
+                conf = ConfWidget.get_default_configuration()
+
+            working_dir = WorkingDirOpts(
+                source=WorkingDirSource.ConfigurationDirectory,
+                path=dirname)
+
+            exec_params = RunExecutionParameters(
+                working_dir=working_dir, executor_params=conf)
+
+            ext_exec_params = ExtendedRunExecutionParameters(
+                uuid=None, name=None, params=exec_params)
+            executor = self.run_executors[last_executor]
+            executor.exec_run_configuration(run_conf, ext_exec_params)
+
+            self.last_executed_per_context |= {(uuid, context)}
+
+            if (context, action_name) in self.re_run_actions:
+                act, _ = self.re_run_actions[(context, action_name)]
+                act.setEnabled(True)
+
+        return anonymous_execution_run
+
+    def run_file(self, selected_uuid=None, selected_executor=None):
+        if not isinstance(selected_uuid, bool) and selected_uuid is not None:
+            self.switch_focused_run_configuration(selected_uuid)
+
+        exec_params = self.get_last_used_executor_parameters(
+            self.currently_selected_configuration)
+
+        first_execution = exec_params['first_execution']
+        display_dialog = exec_params['display_dialog']
+        display_dialog = display_dialog or first_execution
+
+        status, conf = self.edit_run_configurations(
+            display_dialog=display_dialog,
+            selected_executor=selected_executor)
+
+        if status == RunDialogStatus.Close:
+            return
+
+        (uuid, executor_name,
+         ext_params, __) = conf
+
+        if (status & RunDialogStatus.Run) == RunDialogStatus.Run:
+            provider = self.run_metadata_provider[uuid]
+            executor = self.run_executors[executor_name]
+            run_conf = provider.get_run_configuration(uuid)
+
+            context, ext = self.metadata_model.get_metadata_context_extension(
+                uuid)
+            context_name = context['name']
+            context_id = getattr(RunContext, context_name)
```

Comment:
Is this variable `context_id` used for something? Maybe not implemented but to save the last context used or something like what is done for last executed file?

Reply:
We found another way to solve an issue we had, but we never removed this code.